src/xref2/find.ml

```diff
@@ -289,6 +289,20 @@ let extension_in_sig sg name =
     | Signature.TypExt t -> inner t t.Extension.constructors
     | _ -> None)
 
+let extension_decl_in_sig sg name =
+  let rec inner t first = function
+    | ec :: _ when ec.Extension.Constructor.name = name ->
+        Some (`FExt (t, first))
```

Comment:
Note that you are just outputting your input...
Anyway, `extension_in_sig` sufficient to get the info you need, without needing to add the extension decls in the environment!

Reply:
done